contracts/extensions/StreamingPayments.sol

```diff
@@ -328,8 +380,23 @@ contract StreamingPayments is ColonyExtensionMeta {
     require(block.timestamp <= _endTime, "streaming-payments-invalid-end-time");
     require(streamingPayment.startTime <= _endTime, "streaming-payments-invalid-end-time");
 
+    uint256 oldLifetimeClaimable = getAmountClaimableLifetime(_id);
+
     streamingPayment.endTime = _endTime;
 
+    uint256 newLifetimeClaimable = getAmountClaimableLifetime(_id);
+
+    // Unlike when we're setting start time, we need to compare to pseudoAmountClaimedFromStart
+    // in order to determine if the payment is resolved or not.
+    bool wasResolved = streamingPayment.pseudoAmountClaimedFromStart >= oldLifetimeClaimable;
+    bool isResolved = streamingPayment.pseudoAmountClaimedFromStart >= newLifetimeClaimable;
+
+    if (wasResolved && !isResolved) {
```

Comment:
This implementation looks similar to the one above, is there a way to abstract this out somehow?

Reply:
Taken a shot at in a separate commit so you can see what you think :smile: